thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiPresenter.java

```diff
@@ -315,6 +324,32 @@ public boolean isViewAttached() {
         return mState == State.VIEW_ATTACHED;
     }
 
+    /**
+     * Runs the specified action on the UI thread. It only works when a view is attached
+     * <p>
+     * When you are looking for a way to execute code when the view is attached have a look
+     * at {@link #sendToView(ViewAction)}
+     *
+     * @param action the action to run on the UI thread
+     * @throws IllegalStateException when the executor is not available
+     */
+    public void runOnUiThread(@NonNull final Runnable action) {
```

Comment:
Please add some tests for this method.
Also when it is not really usable for "users"... But it is `public`. So we need to take care of it.

Also thinking about to make it just `protected`... Isn't it enough? 🤔